src/osd/osd_types.h

```diff
@@ -424,10 +424,16 @@ struct pg_t {
   unsigned get_split_bits(unsigned pg_num) const;
 
   bool contains(int bits, const ghobject_t& oid) {
-    return oid.match(bits, ps());
+    return
+      ((int64_t)m_pool == oid.hobj.pool ||
+       hobject_t::POOL_TEMP_START-(int64_t)m_pool == oid.hobj.pool) &&
```

Comment:
Shall we use the __is_temp__ wrapper to make sure this is a temp object and then do the pool-id check?

Reply:
yeah, now that we have three places using the fact that the temporary object's pool id is `hobject_t::POOL_TEMP_START - pool`, probably it's time to have a helper function to encapsulate this.

Reply:
What about adding a:
```c++
  int64_t get_pool() const {
    return is_temp() ? (POOL_TEMP_START - pool): pool;
  }
```
wrapper to the __hobject_t__ struct? So we don't bother to check whether it is a temp object...